numba/tests/doc_examples/test_typed_list_usage.py

```diff
@@ -82,6 +88,15 @@ def test_ex_nested_list(self):
             # mylist is now a list of 10 lists, each containing 10 integers
             print(mylist)
 
+            # Just like before, you can directly construct a nested typed-list
+            # from an existing nested Python list
+            py_list = []
+            for i in range(10):
+                l = [j for j in range(10)]
+                py_list.append(l)
+            numba_list = List(py_list)
```

Comment:
I haven't tested this yet because I'm still in the middle of upgrading my system, which will take a while, but I hope this is true =).

Reply:
Perhaps it's a bug you could report 😇:

```
In [3]: List([1, 2, 3])
Out[3]: ListType[int64]([1, 2, 3])

In [4]: List([[1, 2, 3], [4, 5, 6]])
Out[4]: ListType[reflected list(int64)<iv=None>]([[4892186944], [<IPython.lib.pretty.Group object at 0x124ad7040>, <IPython.lib.pretty.Group object at 0x124934cd0>]])

In [5]: List([List([1, 2, 3]), List([4, 5, 6])])
Out[5]: ListType[ListType[int64]]([[1, 2, 3], [4, 5, 6]])
```

Oh and then this happened:

```
In [6]: l = List([[1, 2, 3], [4, 5, 6]])
/Users/vhaenel/git/numba/numba/typed/typedlist.py:66: NumbaPendingDeprecationWarning:
Encountered the use of a type that is scheduled for deprecation: type 'reflected list' found for argument 'item' of function 'impl_append.<locals>.impl'.

For more information visit https://numba.pydata.org/numba-doc/latest/reference/deprecation.html#deprecation-of-reflection-for-list-and-set-types

File "numba/typed/listobject.py", line 596:

    def impl(l, item):
    ^

  l.append(item)
/Users/vhaenel/git/numba/numba/core/ir_utils.py:2031: NumbaPendingDeprecationWarning:
Encountered the use of a type that is scheduled for deprecation: type 'reflected list' found for argument 'item' of function '_append'.

For more information visit https://numba.pydata.org/numba-doc/latest/reference/deprecation.html#deprecation-of-reflection-for-list-and-set-types

File "numba/typed/typedlist.py", line 65:
@njit
def _append(l, item):
^

  warnings.warn(NumbaPendingDeprecationWarning(msg, loc=loc))

In [7]: l
Out[7]: [1]    46853 segmentation fault (core dumped)  ipython
```

Reply:
I have removed the nested typed-list example.